contrib/golang/common/dso/libgolang.h

```diff
@@ -97,35 +97,26 @@ extern "C" {
 
 // go:linkname envoyGoFilterNewHttpPluginConfig
 // github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/http.envoyGoFilterNewHttpPluginConfig
-extern GoUint64
-envoyGoFilterNewHttpPluginConfig(httpConfig* p0); // NOLINT(readability-identifier-naming)
+extern GoUint64 envoyGoFilterNewHttpPluginConfig(httpConfig* p0);
 
 // go:linkname envoyGoFilterDestroyHttpPluginConfig
 // github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/http.envoyGoFilterDestroyHttpPluginConfig
 extern void envoyGoFilterDestroyHttpPluginConfig(GoUint64 id);
 
 // go:linkname envoyGoFilterMergeHttpPluginConfig
 // github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/http.envoyGoFilterMergeHttpPluginConfig
-extern GoUint64
-envoyGoFilterMergeHttpPluginConfig(GoUint64 namePtr,  // NOLINT(readability-identifier-naming)
-                                   GoUint64 nameLen,  // NOLINT(readability-identifier-naming)
-                                   GoUint64 parentId, // NOLINT(readability-identifier-naming)
-                                   GoUint64 childId); // NOLINT(readability-identifier-naming)
+extern GoUint64 envoyGoFilterMergeHttpPluginConfig(GoUint64 namePtr, GoUint64 nameLen,
+                                                   GoUint64 parentId, GoUint64 childId);
```

Comment:
From the preivous source code, seems the parameter name are designed as this type. cc @doujiang24 is there any reason for that?

Reply:
im wondering if we can prefix them - ie `namePtr` -> `goNamePtr` but not sure if that will lose the connection to the go vars

Reply:
IMO, I am incliend to leave these `NOLINT(readability-identifier-naming)` here to let the owner of golang filter make the decision.

And we also needn't to add following config snippet in the clang tidy config file for the golang filter.

```

## TODO(phlax): these are added for Golang filter, figure out how to only apply there
- key: readability-identifier-naming.ParameterIgnoredRegexp
  value: >-
    (bufferLen|bufferPtr|childId|configId|configLen|configPtr|connID|dataPtr|dataSize|dispatcher|endOfStream|endStream|goConnID|headerNum|headerBytes|headerPtr|length|libraryIdLen|libraryIdPtr|nameLen|namePtr|parentId|pluginId|pluginNameLen|pluginNamePtr|pluginPtr|sliceNum)
```

Reply:
the problem is that there is a lot of them - mho is to avoid code stubs/cruft where possible as it adds another layer to decipher - so i generally prefer to do this in config (sadly clang-tidy's config is v v limited)

if we dont move it to config then there is a bunch more of these that will need to added in the code

Reply:
The `.h` file was originally generated by the cgo tool, with some modifications.

We could change `namePtr` to `name_ptr` on the c side, and keep `namePtr` on the Go side, it should work, just loses a little code read-friendly.

I personally think `NOLINT` could be better, but I’m also fine updating them on the c side.

Reply:
im gonna just put back the original NOLINTS and add the missing ones for now